src/dotnet/commands/dotnet-build3/ForwardingApp.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class ForwardingApp
```

Comment:
Having one or two sentences in a comment that describe the high-level of what a class is/does isn't the end of the world.  Rarely does the original intention of a whole class change.

In general I agree with the approach of "strive to make your code as readable as possible without comments".  But I disagree with the approach of "never ever ever use comments".

`NewProcessArgForwardingApp` isn't very readable, IMO.